### PR TITLE
submitting Lesson 5 work

### PR DESCRIPTION
**Pull request by Reem0330**

### CODE REVIEW

**Comment by Reem0330** [comment]
I wasted the full week , just trying out new things .. not sure if I'm going on a circle.. I need help please to figure out what are those errors I'm getting 
for test_mailroom.py .. this is my pytest result
$ pytest
============================= test session starts =============================
platform win32 -- Python 3.7.4, pytest-5.2.1, py-1.8.0, pluggy-0.13.0
rootdir: C:\Users\Usama
collected 6 items

test_mailroom.py FF.FF.                                                  [100%]

================================== FAILURES ===================================
____________________________ test_send_letter_text ____________________________

    def test_send_letter_text():
       letter = '''\n\nDear Reem Alqaysi:\n Thank you for your donation of $222, we appriciate your support to our service. \n MailRoom Team\n'''
>      assert mailroom.thank_you_text('Reem Alqaysi',222) == letter
E      AssertionError: assert None == '\n\nDear Reem Alqaysi:\n Thank you for your donation of $222, we appriciate your support to our service. \n MailRoom Team\n'
E       +  where None = <function thank_you_text at 0x037F18A0>('Reem Alqaysi', 222)
E       +    where <function thank_you_text at 0x037F18A0> = mailroom.thank_you_text

C:\Users\Usama\Documents\SP_Online_PY210\students\ReemAlqaysi\lesson06\test_mailroom.py:17: AssertionError
---------------------------- Captured stdout call -----------------------------


Dear Reem Alqaysi:
 Thank you for your donation of $222, we appriciate your support to our service.
 MailRoom Team

_______________________________ test_new_donor ________________________________

    def test_new_donor():
       fullname = 'Reem Alqaysi'
       mailroom.add_name(fullname)
>      assert fullname in donor_list
E      AssertionError: assert 'Reem Alqaysi' in {'Jan Balard': [600.0, 250.0], 'Jeff Hansen': [450.0, 150.0], 'Joe McHennry': [1500.0, 1500.0], 'Rabi Das': [500.0, 950.0], ...}

C:\Users\Usama\Documents\SP_Online_PY210\students\ReemAlqaysi\lesson06\test_mailroom.py:22: AssertionError
---------------------------- Captured stdout call -----------------------------
{'Jan Balard': [600.0, 250.0], 'Joe McHennry': [1500.0, 1500.0], 'Jeff Hansen': [450.0, 150.0], 'Scott Newman': [100.0, 5000.0], 'Rabi Das': [500.0, 950.0], 'Reem Alqaysi': []}
_______________________________ test_add_amount _______________________________

    def test_add_amount():
        fullname = 'Reem Alqaysi'
        amount = 222
        mailroom.add_amount(fullname,amount)
>       assert donor_list[fullname][:-1] == [amount]
E       KeyError: 'Reem Alqaysi'

C:\Users\Usama\Documents\SP_Online_PY210\students\ReemAlqaysi\lesson06\test_mailroom.py:34: KeyError
_____________________________ test_create_report ______________________________

    def test_create_report():
        report = \
        f'Donor Name        |     Total Given         |Num Gifts                |Average Gift   \n\
        ------------------------------------------------------------------------------------------\n\
        Scott Newman         $      5100.0                2                $      2550.0\n\
        Jeff Hansen          $       600.0                2                $       300.0\n\
        Rabi Das             $      1450.0                2                $       725.0\n\
        Jan Balard           $       850.0                2                $       425.0\n\
        Joe McHennry         $      3000.0                2                $      1500.0\n'

>       assert mailroom.create_report() == report
E       AssertionError: assert None == 'Donor Name        |     Total Given         |Num Gifts                |Average Gift   \n    -------------------------...2                $       425.0\n    Joe McHennry         $      3000.0                2                $      1500.0\n'
E        +  where None = <function create_report at 0x037F1930>()
E        +    where <function create_report at 0x037F1930> = mailroom.create_report

C:\Users\Usama\Documents\SP_Online_PY210\students\ReemAlqaysi\lesson06\test_mailroom.py:46: AssertionError
---------------------------- Captured stdout call -----------------------------

Donor Name        |     Total Given         |Num Gifts                |Average Gift
------------------------------------------------------------------------------------------
Scott Newman          $       5100.0                 2                $      2550.0
Reem Alqaysi          $          222                 1                $       222.0
Jeff Hansen           $        600.0                 2                $       300.0
Rabi Das              $       1450.0                 2                $       725.0
Jan Balard            $        850.0                 2                $       425.0
Joe McHennry          $       3000.0                 2                $      1500.0
========================= 4 failed, 2 passed in 0.14s =========================

for test_send_letter_text()
it's straight forword .. not sure what is the AssertionError is?
also test_add_name.. it's should has same result as test_update_name.. why one passed nut not the other?
for test_create_report .. it should be straight forward.. I counted manually the spaces .. !!
I have less than a month to finish the rest.. I feel kind of frustrated now..
thank you in advance for reviewing the script.. thanks

**Comment by Reem0330** [comment]
ops .. looks like the file and the comments for lesson06 went to lesson05!

File: students/ReemAlqaysi/lesson05/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    print("Bye!")
+    sys.exit()  # exit the interactive script
+
+#send letter to all givers, file name is snake style
+def letter_to_all():
+        for k,v in donor_list.items():
+            amount = k[len(v) - 1]
+            filename = k.replace(' ', '_').replace(',', '') + ".txt"
+            filename = filename.lower()
+            filetext = "Dear {},\n\tThank you for your very kind donation of ${}\n\tIt will be put to very good use.\n\t\t\tSincerely,\n\t\t\t- The Team".format(k,amount)
```
Comment (charlesrobison) [inline students/ReemAlqaysi/lesson05/mailroom.py]: Note the amount isn't printing out correctly

File: students/ReemAlqaysi/lesson05/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                print(key)
+            add_name()
+        elif fullname == "": # if no name entered
+            raise TypeError
+        else:
+            amount = get_amount()
+            if donor_list.get(fullname):
+                donor_list[fullname].append(amount)
+            else:
+                donor_list.update({fullname:(amount)}) # add new name and donations
```
Comment (charlesrobison) [inline students/ReemAlqaysi/lesson05/mailroom.py]: This isn't working quite right - when adding new donors and donations I'm getting errors.

File: students/ReemAlqaysi/lesson05/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+
+#thank you Email formating
+def thank_you_email(fullname, amount):
+    print ("\n\nDear {}:\n Thank you for your donation of ${:2d}, we appriciate your support to our service. \n MailRoom Team\n".format(fullname,amount))
+    main()
+
+
+#create a report that calculate Donor Name, Total Given, Number of donatons, and the avarage amount of thier donations
+def create_report():
```
Comment (charlesrobison) [inline students/ReemAlqaysi/lesson05/mailroom.py]: This function is missing the sorting requirement - per the assignment, donors should be listed by total donation amount given, listed descending.

File: students/ReemAlqaysi/lesson05/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            raise TypeError
+        else:
+            amount = get_amount()
+            if donor_list.get(fullname):
+                donor_list[fullname].append(amount)
+            else:
+                donor_list.update({fullname:(amount)}) # add new name and donations
+        thank_you_email(fullname,amount)
+    except TypeError:
+            print("\nenter a name please\n>>>")
```
Comment (charlesrobison) [inline students/ReemAlqaysi/lesson05/mailroom.py]: Over indented

**Comment by Reem0330** [comment]
Submitted version 2:0 the report was working , not sure what happened, a line got deleted from the file in somehow .. sorry for that

**Pull request merged**